apps/web/src/components/form/ListboxSelect.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { Fragment, useCallback } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown, faCheck, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { ListboxButton, ListboxOptions, ListboxOption, Transition } from '@headlessui/react';
+
+import { OptionType } from '@components';
+
+type MultiItemListboxProps = {
+  id: string;
+  options: OptionType[];
+  selected: OptionType[];
+  handlePreviousSelect: (value: OptionType) => void;
+  isDisabled?: boolean;
+  menuPlacement?: 'bottom' | 'top';
+};
+
+const listboxStyles = (
+  isDisabled: boolean | undefined,
+  optionsLength: number,
+  selectedLength?: number,
+) => {
+  const disabledStyles =
+    isDisabled || optionsLength === 0
+      ? 'bg-gray-300 '
+      : 'border-b-2 border-black bg-gray-100 hover:border-gray-400';
```

Comment:
Nit - To improve readability, maybe either add a comment above this line going over what it does, or replacing the inline css with constants with names that reflect their function. Also there is a trailing space in bg-gray-300.